### PR TITLE
Update ruby_services.md

### PR DESCRIPTION
**Pull request by cintamani**
Add team guidelines for tests in regard to empty lines dividing logically the AAA test blocks and the standard status of factories.

### CODE REVIEW

File: teams/ruby_services.md [hunk 1]
```diff
@@ -110,7 +110,7 @@ For example
 ```ruby
 factory :story do
   title { "My awesome story" }
-
+
```
Comment (cintamani) [inline teams/ruby_services.md]:
```suggestion

```

File: teams/ruby_services.md [hunk 2]
```diff
@@ -158,6 +157,26 @@ end
 test_story = build(:story, :published_but_not_in_print)
 ```
 
+#### Assemble - Assert - Act
+
+We agree that tests should divide their Assert, Assemble and Act logic using empty lines. For example
+
+```
```
Comment (cintamani) [inline teams/ruby_services.md]:
```suggestion
```ruby
```

File: teams/ruby_services.md [hunk 3]
```diff
@@ -110,7 +110,6 @@ For example
 ```ruby
 factory :story do
   title { "My awesome story" }
-
   trait :published do
```
Comment (cintamani) [inline teams/ruby_services.md]:
```suggestion

  trait :published do
```

File: teams/ruby_services.md [hunk 4]
```diff
@@ -129,16 +129,15 @@ test_story = build(:story, :published, in_print: false)
 
 The issue we have found is that as time goes on, the required conditions of the object may change, and then there is contention as to whether the existing trait gets updated or the test code.
 
-To avoid this our agreed pattern is that *all object setup should be in a trait*.
+We have then decided that every factory will have a base status of data. Any test that requires specific data to be there which are different from the `default` status of the object will then have to define those in a Trait.
```
Comment (cintamani) [inline teams/ruby_services.md]:
```suggestion
We have then decided that every factory will have a base status of data. Any test that requires specific data to be there which are different from the `default` status of the object will then have to define those in a Trait.
```

File: teams/ruby_services.md [hunk 5]
```diff
@@ -129,16 +129,15 @@ test_story = build(:story, :published, in_print: false)
 
 The issue we have found is that as time goes on, the required conditions of the object may change, and then there is contention as to whether the existing trait gets updated or the test code.
 
-To avoid this our agreed pattern is that *all object setup should be in a trait*.
+We have then decided that every factory will have a base status of data. Any test that requires specific data to be there which are different from the `default` status of the object will then have to define those in a Trait. 
+For example, a test for the presence of a specific `title` should define a new trait with the expected title setup and test for it rather than use the base object. The idea is that basic data allow validations of the object through the different transition states but **will not** guarantee that the data will be consistent over time unless placed in a trait.
```
Comment (cintamani) [inline teams/ruby_services.md]:
```suggestion
For example, a test for the presence of a specific `title` should define a new trait with the expected title setup and test for it rather than use the base object. The idea is that basic data allow validations of the object through the different transition states but **will not** guarantee that the data will be consistent over time unless placed in a trait.
```

File: teams/ruby_services.md [hunk 6]
```diff
@@ -129,16 +129,15 @@ test_story = build(:story, :published, in_print: false)
...
 ```ruby
 factory :story do
   title { "My awesome story" }
-
-  trait :published do
-    published { true }
-  end
-
+  published { true }
+  in_print { true }
```
Comment (cintamani) [inline teams/ruby_services.md]:
```suggestion
  in_print { true }
```

File: teams/ruby_services.md [hunk 7]
```diff
@@ -129,16 +129,15 @@ test_story = build(:story, :published, in_print: false)
...
 factory :story do
   title { "My awesome story" }
-
-  trait :published do
-    published { true }
-  end
-
+  published { true }
+  in_print { true }  
+
```
Comment (cintamani) [inline teams/ruby_services.md]:
```suggestion

```

File: teams/ruby_services.md [hunk 8]
```diff
@@ -129,16 +129,14 @@ test_story = build(:story, :published, in_print: false)
 
 The issue we have found is that as time goes on, the required conditions of the object may change, and then there is contention as to whether the existing trait gets updated or the test code.
 
-To avoid this our agreed pattern is that *all object setup should be in a trait*.
+We have then decided that every factory will have a base status of data. Any test that requires specific data to be there which are different from the `default` status of the object will then have to define those in a Trait.
```
Comment (irisfaraway) [inline teams/ruby_services.md]:
```suggestion
Every factory should include the essential data by default. If a test needs specific data that differs from the `default` status of the object, it should use a trait to define the data.
```

I thought "a base status of data" could be a bit clearer, so tried to reword it and ended up with this – what do you think?

Comment (cintamani) [reply teams/ruby_services.md]: Sounds much better!

**Comment by cintamani** [comment]
Closing this so I can open 2 different PRS per change
